### PR TITLE
Reorganize SCSS files

### PR DESCRIPTION
**Pull request by a-dg**
This reorganizes the SCSS files in preparation for incorporating component styles from Classic.

### CODE REVIEW

**Comment by Ollo** [comment]
We will need to test this update in League Frontend. cc/ @mikereinmiller

File: src/fonts/index.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@import "museo";
+// @import "open-sans";
```
Comment (williamdingwall) [inline src/fonts/index.scss]: Not sure why this is commented out - we can include

File: src/teamsnap-ui.scss [hunk 2]
```diff
@@ -1,3 +1,5 @@
+@import "config/index";
 @import "base/index";
+@import "fonts/index";
```
Comment (williamdingwall) [inline src/teamsnap-ui.scss]: minor nit, probably doesn't matter functionally, but what do you think about moving fonts above config?

File: src/utils/size.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-@import "../base/config.scss";
+@import "../config/config.scss";
```
Comment (williamdingwall) [inline src/utils/size.scss]: We'll need to do this for all the components too..  `@import "../config/config.scss";` needs to be included for each one so they can all be imported and composed upon individually within react.

File: src/base/index.scss [hunk 4]
```diff
@@ -1,4 +1,4 @@
-@import "functions";
-@import "mixins";
-@import "config";
+@import "reset";
 @import "typography";
+@import "lists";
+@import "tables";
```
Comment (saralohr) [inline src/base/index.scss]: Should lists and tables be included? I don't see files for these.

**Comment by saralohr** [comment]
> @saralohr any thoughts on this structure with the new config and base styles folders?

This seems fine to me.

File: src/fonts/index.scss (hunk 1 above)
Comment (a-dg) [inline src/fonts/index.scss]: I had left it out temporarily, as I didn't know if all of the additional character sets (Cyrillic, Greek) were necessary. I'll uncomment this.

File: src/base/index.scss (hunk 4 above)
Comment (a-dg) [inline src/base/index.scss]: Good catch, those will be in a separate PR. Thanks!

File: src/teamsnap-ui.scss (hunk 2 above)
Comment (a-dg) [inline src/teamsnap-ui.scss]: Done, I agree that things with no dependencies are better off being imported first.

File: src/utils/size.scss (hunk 3 above)
Comment (a-dg) [inline src/utils/size.scss]: It seems like anything previously importing `base/index.scss` should import `config/index.scss`, as that's going to pull in the functions and mixins, as well as the new `base/index.scss` to get the reset and typography styles. Does that seem right?

Comment (saralohr) [inline src/utils/size.scss]: Should this be `config/index.scss`?

File: src/base/reset.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
+@import "../config/index.scss";
```
Comment (saralohr) [inline src/base/reset.scss]: Does this file need to import the config?

File: src/config/config.scss [hunk 6]
```diff
@@ -140,7 +140,7 @@ $su-xxxsmall:   scaleGrid(-3) !default; // 1px
 $su-xxsmall:    scaleGrid(-2) !default; // 2px
 $su-xsmall:     scaleGrid(-1) !default; // 4px
 $su-small:       scaleGrid(1) !default; // 8px
-$su-default:     scaleGrid(2) !default; // 16px
+$su-medium:     scaleGrid(2) !default; // 16px
```
Comment (saralohr) [inline src/config/config.scss]: 👍 This makes more sense to me. We should probably update in Classic too.

File: src/base/reset.scss (hunk 5 above)
Comment (williamdingwall) [reply src/base/reset.scss]: not currently, could potentially in the future if htere ever is a specific value needed, unless it's intended purpose will always be to set things to "0".. but just went with it for consistency.

Comment (williamdingwall) [reply src/base/reset.scss]: I'm fine with cutting here

File: src/config/config.scss (hunk 6 above)
Comment (williamdingwall) [reply src/config/config.scss]: I can put in a follow up story for that... think there's a few other spots with the functions that need updating in classic too

File: src/base/reset.scss (hunk 5 above)
Comment (saralohr) [reply src/base/reset.scss]: I guess it's a matter of whether this file (or typography) are being imported directly into the react components? Not sure how these more general base styles are being handled currently, or if they are just relying on Classic for these things. If they are being pulled in in some way, then including it for consistency makes sense...

Comment (williamdingwall) [reply src/base/reset.scss]: I could see a scenario where typography is brought in by itself for various projects... reset less so.  Adding that config in the base files is more about future projects like the marketing site

Comment (saralohr) [reply src/base/reset.scss]: Alrighty, well I'm fine either way. Take it or leave it.

File: src/utils/size.scss (hunk 3 above)
Comment (williamdingwall) [reply src/utils/size.scss]: ah, yep, updated
